### ISSUE TITLE
is there any undocumented function to recalculate the face normals for consistency?

### ISSUE THREAD

**Issue by xiaoxTM** [facebookresearch/pytorch3d#494]
## 🚀 Feature

recalculate the face normals for consistency
NOTE: Please look at the existing list of Issues tagged with the label ['enhancement`](https://github.com/facebookresearch/pytorch3d/issues?q=label%3Aenhancement). **Only open a new issue if you do not see your feature request there**.

## Motivation

sometimes there are wrong ordered face indices in mesh files (e.g., OFF file store vertex index to represent face). Directly rendering them generates wrong images that have non-interpolation-like texture. It will be great if there are functions to make the face normals consistent (i.e., all pointing outside or inside of a object), so that the rendered images have better looks.
## Pitch

the function may looks like:
```python
meshes.make_face_normals_consistency()
```
NOTE: we only consider adding new features if they are useful for many users.

**Comment by russelldc** [comment]
I would really love this as well.
Is there a common way to do this "manually" with PyTorch3D's Mesh objects?

**Comment by martinpflaum** [comment]
Hi thats actually really simple. You can calculate the normals for every surface relativly easily. But there are a couple of things to be aware:

1.  The direction of the normals depends on the order the vectors that descripe the surface. So given 3 vectors x,y,z that descripe a surface. A surface that is constructed with x,y,z is not equal a surface that is constructed with z,y,x. The direction a Surface is visible only depends on the ordering of the verticies! Not on the surface. If the ordering of you verticies is messed you can't compute the correct normal.
2. Normals are usally used to smooth out edges. We can only recalculate normals that will make every edge extremly sharp.

```python
def calc_norm(points):
    """
    https://math.stackexchange.com/questions/305642/how-to-find-surface-normal-of-a-triangle

    input is 1 face - 3x3 tensor 
    """
    a,b,c = points
    v = b - a
    w = c - a
    dir = torch.cross(b - a, c - a)
    norm = dir/torch.norm(dir)
    return norm
```
This is the code for validating that the above is correct.
``` python 
#%%
from mpl_toolkits.mplot3d import Axes3D
from matplotlib.patches import FancyArrowPatch
import matplotlib.pyplot as plt
import numpy as np
from mpl_toolkits.mplot3d import proj3d

def plot_surface_and_normal(a,b,c,v):
        
    x = [a[0],b[0],c[0]]
    y = [a[1],b[1],c[1]]
    z = [a[2],b[2],c[2]]
    print(x,y,z)
    class Arrow3D(FancyArrowPatch):
        def __init__(self, xs, ys, zs, *args, **kwargs):
            FancyArrowPatch.__init__(self, (0,0), (0,0), *args, **kwargs)
            self._verts3d = xs, ys, zs

        def draw(self, renderer):
            xs3d, ys3d, zs3d = self._verts3d
            xs, ys, zs = proj3d.proj_transform(xs3d, ys3d, zs3d, renderer.M)
            self.set_positions((xs[0],ys[0]),(xs[1],ys[1]))
            FancyArrowPatch.draw(self, renderer)
    n_radii = 8
    n_angles = 36

    radii = np.linspace(0.125, 1.0, n_radii)
    angles = np.linspace(0, 2*np.pi, n_angles, endpoint=False)[..., np.newaxis]

    a = Arrow3D([np.mean(x), v[0]], [np.mean(y), v[1]], [np.mean(z), v[2]], mutation_scale=20, lw=3, arrowstyle="-|>", color="r")
    ax = plt.figure().add_subplot(projection='3d')

    ax.plot_trisurf(x, y, z, linewidth=0.2, antialiased=True,alpha = 0.5)

    ax.add_artist(a)
    ax.scatter(v[0],v[1],v[2])
    

import torch

x = torch.tensor([0.,0.,0.])
y = torch.tensor([1.,0.,0.])
z = torch.tensor([0.,1.,0.])



def calc_norm(points):
    """
    https://math.stackexchange.com/questions/305642/how-to-find-surface-normal-of-a-triangle

    input is 1 face - 3x3 tensor 
    """
    a,b,c = points
    v = b - a
    w = c - a
    dir = torch.cross(b - a, c - a)
    norm = dir/torch.norm(dir)
    return norm




normal = calc_norm((x,y,z))

x,y,z,normal = x.numpy(),y.numpy(),z.numpy(),normal.numpy()

plot_surface_and_normal(x,y,z,normal+np.mean([x,y,z],axis=0))
plt.show()
# %%
```

**Comment by martinpflaum** [comment]
you could load the file via trimesh and then use the function
trimesh.repair.fix_inversion
